### PR TITLE
Add releasing notes

### PR DESCRIPTION
**Pull request by gadomski**
**Related Issue(s):** n/a

**Description:** Adds a document describing the release process.

**PR Checklist:**

- [x] Code is formatted
- [x] Tests pass

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+4. Update [CHANGELOG.md](CHANGELOG.md).
+   1. Add a new header under "Unreleased" with the new version and the date, e.g. `## [vX.Y.Z] - YYYY-MM-DD`.
+   2. Audit the changelog section to ensure all changes are captured.
+   3. Add a new link reference at the bottom of the changelog with the new version.
+      Follow the format from the previous version links.
+   4. Update the "Unreleased" link to reference your new version.
+5. (optional) Build the package locally and inspect its contents: `pip install build && python -m build`
+6. Open a pull request for your `release/vX.Y.Z` branch against the appropriate branch (either `main` or a version branch, e.g. `v0.3`).
+   Include a section in the pull request description with the CHANGELOG contents for this version.
+7. After pull request merge, create an annotated tag for your version, e.g. `git tag -a vX.Y.Z`.
```
Comment (philvarner) [inline RELEASING.md:16]: is there a reason we tag and then do the GH release instead of doing the GH release (which tags) first?

Comment (gadomski) [reply RELEASING.md:16]: GH releases don't create annotated tags, so `git describe` doesn't behave nicely.

Comment (philvarner) [reply RELEASING.md:16]: oh, I thought they did. I'm going to have to stop doing that then.

Comment (gadomski) [reply RELEASING.md:16]: Yeah, I just tested here: https://github.com/gadomski/gadomski/tree/v0.0.1. Not annotated.

**Review by philvarner — APPROVED** [review]

**Pull request merged**